src/common/mfu_daos.c

```diff
@@ -562,6 +578,8 @@ static int daos_connect(
         /* everything looks good so far */
         valid = 1;
     }
+    if (connect_all_ranks)
+        goto skip_bcast;
 
 bcast:
     /* broadcast valid from rank 0 */
```

Comment:
@dsikich Since some error conditions above have `goto bcast`, we might want to instead do this:
```suggestion

bcast:
    if (connect_all_ranks)
        goto skip_bcast;
    /* broadcast valid from rank 0 */
```